### PR TITLE
WIP Selenium tests for literature and author forms

### LINKED ISSUE #1593: Selenium tests for literature and author forms
# Literature suggestion form

1) Access http://localhost:5000/submit/literature/create
2) Test the form validators. Form is defined in [this file](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/literaturesuggest/forms.py) and an example of [field with validator](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/literaturesuggest/forms.py#L234)
3) Submit the form and get success page.

Now you should see in the console logs that the worker starts doing some computation (executing the workflow) and after some seconds the record should be in the Holding Pen. Let's see if that is true:

4) Access http://localhost:5000/holdingpen/ and check that the dashboard shows the new submission

screen shot 2016-09-23 at 11 04 35 am

5) Click on the Halted link and you will get to the Holding Pen listing:

screen shot 2016-09-23 at 11 04 58 am

6) Click on the holding pen entry and do some assertions on the detailed page, e.g, check that the reject, accept and core buttons are there and check that the metadata you filled in the form is there: title, authors, subject and so on.

7) Click on Accept and make sure you get the right message on the page:

screen shot 2016-09-23 at 11 08 56 am
# Author suggestion form

1) Go to http://localhost:5000/submit/author/create

2) Fill the form and test validators. Form is defined in [this file](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/authors/forms.py)

3) Submit and get success page.

4) Go to http://localhost:5000/holdingpen/ and confirm the dashboard shows correct info.

screen shot 2016-09-23 at 11 14 21 am

5) Click on Halted to see the entry that was just submitted

screen shot 2016-09-23 at 2 48 52 pm

6) Go into the detailed page and make sure the 4 action buttons are there and that the metadata corresponds to the metadata entered in the form.

screen shot 2016-09-23 at 2 49 23 pm

7) Clicking on Accept, Reject or Core should show a success message.

8) Clicking on Review Submission should go back to the original form.
- There should be 2 extra fields in the form (that are only in the review): inspireid and bai

9) Try submitting the review form.
# Author update form

Same as author suggestion form until accessing the detailed page in the Holding Pen.

The detailed page should not have any action button.

### PR DESCRIPTION
**Pull request by rikirenz**
Closes #1593

### CODE REVIEW

File: tests/acceptance/test_literature_suggestion_form.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+from selenium.common.exceptions import ElementNotVisibleException
+from selenium.common.exceptions import WebDriverException
+
+
+def test_literature_create_article_journal_manually(selenium, login):
+    """Submit the form for article creation from scratch"""
+    selenium.get(os.environ['SERVER_NAME'] + '/submit/literature/create')
+    _hide_title_bar(selenium)
+    selenium.find_element_by_id("skipImportData").click()
+    WebDriverWait(selenium, 10).until(EC.text_to_be_present_in_element((By.ID, "form_container"),'Type of Document'))
```
Comment (jacquerie) [inline tests/acceptance/test_literature_suggestion_form.py]: I read [here](http://www.seleniumhq.org/docs/04_webdriver_advanced.jsp) that you should not mix implicit and explicit waits... what happens if you remove the implicit wait?

Comment (jacquerie) [inline tests/acceptance/test_literature_suggestion_form.py]: I was just curious, does it really work?!

You could remove it in the same commit in which you add the explicit waits, with a note in the commit message about why you did it.

File: tests/acceptance/test_literature_suggestion_form.py [hunk 2]
```diff
@@ -90,6 +90,19 @@ def test_format_input_DOI(selenium, login):
     assert "The provided DOI is invalid - it should look" not in selenium.page_source
 
 
+def test_basic_info_autocomplete_affilation(selenium, login):
+    """Test the autocompletion for the affilation in the basic info section"""
+    selenium.get(os.environ['SERVER_NAME'] + '/submit/literature/create')
```
Comment (kaplun) [inline tests/acceptance/test_literature_suggestion_form.py]: Eheh. Is there a quick non-hackish better way that @rikirenz could apply?

**Comment by jacquerie** [comment]
I think we should start merging things from this PR today.

**Comment by rikirenz** [comment]
I am going to refactor this branch in according to some guidelines that I discovered in this video: https://www.youtube.com/watch?v=DO8KVe00kcU.

Some errors that I have done and why these are errors:
* I have recorded tests. Usually these way of write tests put the id/xpath/class of the elements directly inside the tests. If the UI will change all the tests, that are using that specific ID, are going to fail. 

* I have replicated a lot of code in order to interact with the web app's components (and most of the components are the same almost everywhere). It means that if the UI change (for instance: we change a checkbox in a opt button) it will be necessary change the code in several points.

* I have tried to create simple and atomic tests. Anyway I think that the code is not very readable. The test code should be explain what is testing and why is failing. I think now this is not always true.

These are the problems... and this is the solution that I would like to implement:

Architecture (each layer interacts with the layer immediately below): 
* tests
* inspire_selenium_framework
* selenium
* web_app


## This is an <h2> tag
## This is an <h2> tag
## This is an <h2> tag

**Comment by rikirenz** [comment]
I am going to refactor this branch in according to some guidelines that I discovered in this video: https://www.youtube.com/watch?v=DO8KVe00kcU.

Some errors that I have done and why these are errors:
* I have recorded tests. Usually these way of write tests put the id/xpath/class of the elements directly inside the tests. If the UI will change all the tests, that are using that specific ID, are going to fail. 

* I have replicated a lot of code in order to interact with the web app's components (and most of the components are the same almost everywhere). It means that if the UI change (for instance: we change a checkbox in a opt button) it will be necessary change the code in several points.

* I have tried to create simple and atomic tests. Anyway I think that the code is not very readable. The test code should be explain what is testing and why is failing. I think now this is not always true.

These are the problems... and this is the solution that I would like to implement:

Architecture (each layer interacts with the layer immediately below): 
* tests
* inspire_selenium_framework
* selenium
* web_app

Where the framework should be structured in this way:
<table>
  <tr>
    <td align="center" colspan="2">Workflows</td>
  </tr>
  <tr>
    <td align="center" colspan="2">Pages</td>
  </tr>
  <tr>
    <td align="center" >Navigation</td>
   <td align="center">UI utilities</td>
  </tr>
  <tr>
    <td align="center" colspan="2">selenium</td>
  </tr>
</table>

**Comment by rikirenz** [comment]
The tests should not pass on travis. But locally I have tried to rebase my branch on the branch of @david-caro (`hepcrawl_integration`) and they were working. It means that when it will be merged I will rebase my branch on master and everything should work properly.

**Comment by rikirenz** [comment]
Guys the PR is complete. I am rebasing stuff on a local branch to make the tests more maintainable and readable. Anyway if you wanna start to merge this, all the tests are working. @jacquerie @kaplun of course comments are welcome even though I am rebasing it.

**Comment by jacquerie** [comment]
Superseded by #1730.

**Pull request closed without merging**